### DISCUSSION TITLE
patterns/builder/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#55] [Announcements]
# patterns/builder/

Intent Separate the construction of a complex object from its representation so that the same construction process can create different representations. Explanation Real-world example Imagine a character generator for a role-playing game. The easiest option is to let the computer create the character for you. If you want to manually select the character details like profession, gender, hair color, etc. the character generation becomes a step-by-step process that completes when all the selections are ready.

https://java-design-patterns.com/patterns/builder/

**Comment by galessandro** [comment]
in Hero's class constructor, how can it access to Builder properties if they are marked as private?

**Reply by iluwatar** [reply]
This is how Java's nested classes work
